plugins/thunderbolt/fu-thunderbolt-image.c

```diff
@@ -94,7 +94,7 @@ read_location (const FuThunderboltFwLocation  *location,
 		g_set_error (error,
 			     FWUPD_ERROR, FWUPD_ERROR_READ,
```

Comment:
I assume you are talking about this error code.
This function is generic both for the controller itself and the image, this is why I chose the more generic `FWUPD_ERROR_READ` instead of `FWUPD_ERROR_INVALID_FILE`, as the latter doesn't make any sense if the read from the controller is what failed.
We can add a flag to `FuThunderboltFwObject` to tell if this is a controller or image file and return the relevant error code depending on this flag.

Reply:
That also makes sense. I am wondering what would be a good error code if the verification for controller itself fails. None really seem to fit properly.

Reply:
>  if the verification for controller itself fails

Not sure what it means. The controller is always valid, otherwise it wouldn't work (or would be in safe-mode). If there is a read error, this looks like the relevant error.
Maybe you meant if we find there is no DROM or DRAM sections when they're expected? [line 538](https://github.com/hughsie/fwupd/pull/214/files/c65072211c75d31f6ca1b353313009b513df038f#diff-0d02b79a376b5c204ecf747b7877cbfaL538)
I used `FWUPD_ERROR_READ` because I can't imaging any other reason for this to happen. Wrong image flashed to the controller by the OEM on manufacturing line?

Reply:
Sorry, yeah, that is what I meant. Seems indeed like a corner case, maybe not worth thinking too much about.